### PR TITLE
(baobab) Update CI Helm Lint workflow to include pull request events

### LINKED ISSUE #56: Write CI script for helm chart
@nick-bisonai found this error while deploying with ArgoCD.
Image

For next time, write GitHub Actions to execute `helm lint` and only mergable the code when passed this actions.

### PR DESCRIPTION
**Pull request by jay-bisonai**
This pull request updates the CI Helm Lint workflow to include pull request events. It adds a new job called "lint". The job checks out the repository and runs Helm lint on all changed files. If any service fails the linting process, the job outputs the names of the failed services. This update ensures that the CI Helm Lint workflow is triggered for pull request events and provides information about failed services.

### CODE REVIEW

File: .github/workflows/ci.helm-lint.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: CI Helm Lint
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+    branches:
+      - main
+      - gcp-cypress-prod
```
Comment (martinkersner) [inline .github/workflows/ci.helm-lint.yaml:10]: I think we want to run it only on `gcp-baobab-prod` 🤔

**Review by martinkersner — APPROVED** [review]
LGTM! Looks super useful!

**Review by nick-bisonai — APPROVED** [review]
LGTM!

File: .github/workflows/ci.helm-lint.yaml (hunk 1 above)
Comment (jay-bisonai) [reply .github/workflows/ci.helm-lint.yaml]: Also I think so, thank you for suggested. 🙏

**Pull request merged**